### ISSUE TITLE
settings in templates

### ISSUE THREAD

**Issue by nerdoc** [IlyaSemenov/django-modelsettings#6]
thanks for this great module.

I needed some settings in templates, like 

```
{% if settings.homepage.show_title %}
  {{ title }}
{% endif %}
```

So I ended up in creating a context_processor that fetches the settings and returns a `settings`dict to the context.

There are a few things that could be improved here, maybe I should file them as separate bugs, but they are all related more or less, so I decided to put that in one here first:

<s>* [ ] CollectedSettings are the way to go here IMHO. This (and collect_settings) should use the cache framework.</s>
<s>* [ ] CollectedSettings should have settings as class attr, not instance attr, to let more instances use the same fetched settings during a request.</s> -> in separate issue
<s>* [ ] CollectedSettings needs some documentation (string)</s> -> in separate issue
* [x] The context processor should be added to modelsettings for convenience.

If you agree with me, I could try to do that work and make a PR, just tell me what you think about it.

Maybe CollectedSettings is an internal class that shouldn't be used anyway, but `dbsettings.settings` instead. But the caching problem remains.

**Comment by IlyaSemenov** [comment]
@nerdoc Honestly it's been many years since this was written and today I'm not even sure what CollectedSettings are... I looked up the source but still I'm not 100% confident in what I see. 😃 

I agree that this should benefit from caching and the built-in context processor. I was using an ad-hoc context processor every time in my projects anyway.

I'm not inclined that there should be anything hidden. What is the problem that it's supposed to solve? Templates are similarly the source code just in a different language. We don't "hide" anything from "normal" models at templates, why would hide data from settings?

**Comment by IlyaSemenov** [comment]
So yes please come up with the PRs, if I see that you deliver code with reasonable quality, I can simply add you to collaborators (here and on pypi).

**Comment by nerdoc** [comment]
> I'm not inclined that there should be anything hidden. What is the problem that it's supposed to solve? Templates is similarly source code just in a different language. We don't "hide" anything from "normal" models at templates, why would we hide data from settings?

Yes, that's true. Wrong thought, and not  at all interesting at the Settings level anyway, more on fields. But yes, for get that.

**Comment by nerdoc** [comment]
I'll do the CollectedSettings in another PR, should have separated that in the first place...

**Comment by nerdoc** [comment]
Is there a chance for a fix release 0.5.2? Would appreciate it for not having to install forks all the time ;-)
